src/nvim/eval.c

```diff
@@ -11686,14 +11691,70 @@ static void f_jobstart(typval_T *argvars, typval_T *rettv, FunPtr fptr)
         return;
       }
     }
+    dictitem_T *item;
+    // or new_env
+    item = tv_dict_find(job_opts, S_LEN("env"));
+    if (item) {
+      size_t env_size = 0; // len ?
+      size_t i = 0;
+      // size_t orig_env_size = 0;
+      if (item->di_tv.v_type != VAR_DICT) {
+        EMSG2(_(e_invarg2), "Expected a dictionary for 'env'");
```

Comment:
It is better to use a new error number, never use `e_invarg` and avoid `e_invarg2`. And English phrases are supposed to be translatable.

Reply:
@jamessan zyx's comment is still relevant. I would just change it to `EMSG2(e_invarg2, "env")` for now.

> way to choose a free error code ? or should I just search & increase an number until I can't find a match ?

https://github.com/neovim/neovim/pull/8155